pkgs/applications/misc/waybar/default.nix

```diff
@@ -21,6 +22,11 @@
 
     nativeBuildInputs = [
       meson ninja pkgconfig scdoc wrapGAppsHook cmake
+    ] ++ stdenv.lib.optional withMediaPlayer playerctl;
+
+    propagatedBuildInputs = stdenv.lib.optionals withMediaPlayer [
+      gobject-introspection
+      python38Packages.pygobject3
     ];
 
     buildInputs = with stdenv.lib;
```

Comment:
you need to conditionally add `glib` because it uses it via gobject-introspection https://github.com/Alexays/Waybar/blob/master/resources/custom_modules/mediaplayer.py#L9

not sure where the playerctl module comes from